server/text_generation_server/models/__init__.py

```diff
@@ -99,7 +99,10 @@ def get_model(
         else:
             return Galactica(model_id, revision, quantize=quantize)
 
-    if "bigcode" in model_id:
+    config = AutoConfig.from_pretrained(model_id, revision=revision)
+    model_type = config.model_type
+
+    if model_type == "gpt_bigcode": 
```

Comment:
The original santacoder weights have model_type `gpt2`, some people might still be using those with this https://huggingface.co/bigcode/santacoder

Reply:
Maybe add `or model_id.startswith("bigcode/")`?

Reply:
We can have both. We deployed this morning new fine tuned version that do require the `gpt_bigcode` thing.

We'll see if we can update those old models too (using the config is the only thing that doesn't depend on the model name ;).

Reply:
This logic does not work for bigcode/santacoder because it doesn't have the correct model type but also because it requires `trust_remote_code = True` to load the config.
Beside duplicating this block, idk what's the best solution here.